woocommerce/woocommerce
# Enhance Build Watching

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This pull request adds a new `watch:build:project` script that is intended to mirror the behavior of `build:project` except for watching. This allows us to set up project-specific watch commands, such as those taking better advantage of `webpack --watch`. The `watch:build` command has been updated to replicate the `build` command and cascades through children.

Closes #42012.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter='@woocommerce/plugin-woocommerce' build` to populate any caches
2. Run `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build`
3. Make changes to any of WooCommerce's dependent packages and watch that it cascades properly.
4. Ensure that any changes are reflected in updates to code in the development environment.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
